website/other/bezier-rs-demos/src/main.ts

```diff
@@ -32,31 +17,63 @@ declare global {
 	}
 }
 
+window.document.title = "Bezier-rs Interactive Documentation";
+
 window.customElements.define("bezier-demo", BezierDemo);
 window.customElements.define("bezier-demo-pane", BezierDemoPane);
 window.customElements.define("subpath-demo", SubpathDemo);
 window.customElements.define("subpath-demo-pane", SubpathDemoPane);
 
-const bezierDemos = document.getElementById("bezier-demos");
-(Object.keys(bezierFeatures) as BezierFeatureName[]).forEach((featureName) => {
+function renderBezierPane(featureName: BezierFeatureKey, container: HTMLElement | null): void {
 	const feature = bezierFeatures[featureName];
 	const demo = document.createElement("bezier-demo-pane");
 
 	demo.setAttribute("name", featureName);
 	demo.setAttribute("demoOptions", JSON.stringify(feature.demoOptions || {}));
 	demo.setAttribute("triggerOnMouseMove", String(feature.triggerOnMouseMove));
 	demo.setAttribute("chooseComputeType", String(feature.chooseComputeType));
-	bezierDemos?.append(demo);
-});
+	container?.append(demo);
+}
 
-const subpathDemos = document.getElementById("subpath-demos");
-(Object.keys(subpathFeatures) as SubpathFeatureName[]).forEach((featureName) => {
+function renderSubpathPane(featureName: SubpathFeatureKey, container: HTMLElement | null): void {
 	const feature = subpathFeatures[featureName];
 	const demo = document.createElement("subpath-demo-pane");
 
 	demo.setAttribute("name", featureName);
 	demo.setAttribute("sliderOptions", JSON.stringify(feature.sliderOptions || []));
 	demo.setAttribute("triggerOnMouseMove", String(feature.triggerOnMouseMove));
 	demo.setAttribute("chooseComputeType", String(feature.chooseComputeType));
-	subpathDemos?.append(demo);
-});
+	container?.append(demo);
+}
+
+const pathname = window.location.pathname;
+const splitPathName = pathname.split("/");
```

Comment:
You want to use `window.location.hash`.

Reply:
To implement the behaviour that I desire here, I intend to use `pathname` not `hash`. This part of the code is used to determine which of the examples to render, whereas `hash` will be used to jump to one of the rendered example.

To further explain the behaviour here: This will parse a url (ex. `http://localhost:8080/bezier/trim`) and then render only the specified example (ex. the bezier trim). This is intended to facilitate embedding a single example in an iframe for the rust docs.

![image](https://user-images.githubusercontent.com/30204508/213080230-895d7435-76cf-45c9-b637-116836ccc1ae.png)

Reply:
I'm confused about the difference between the two things you're referring to. In your "bezier trim" example, what is the "bezier" part?

Reply:
"bezier" is used to differentiate between showing the beizer vs subpath versions of a function

Reply:
I see. Those will both have to be part of the same hash fragment then. Routing based on the actual URL requires special treatment from the web server, which we can't do for the deployed website. It just so happens that Vue/Webpack is letting you visit the page (I'm not actually sure why it's configured to let you do that), but that's not true in production once the site is built and not served through Webpack's dev server.

Reply:
My intentions are that `localhost:8080/` will render all the examples for both beziers and subpaths. On this page, the hash will be used to jump to each example (ex. `localhost:8080/#bezier_trim`). If we want the page to render only one example, this can be done by going to the corresponding url with the following pattern: `localhost:8080/:bezierOrSubpath/:exampleKey` where `beizerOrSubpath` is either "bezier" or "subpath, and `exampleKey` is the corresponding example key in the `bezierFeatures` and `subpathFeatures` files.

Reply:
Based on your other comment, there seems to be a difference between what I intended vs what you were requesting.

Are you sure that you want a url with a hash to hide all the other examples? It seems like common practice for this is to scroll to the entity, while keeping the rest of the page the same.

Reply:
Yeah, see above about why we can't do that, sorry. If you want to make a routing scheme, you can implement that with the hash fragment only. Perhaps something like `#bezier/trim` and `#bezier/trim/solo` could be used for normal navigation vs. soloing a single entry for the iframe. You'll need your own logic to achieve that latter case by checking the end of the hash fragment and hiding all the others in that case. I suggested using CSS's `:target` selector earlier but for performance reasons I'd actually suggest filtering the list down to the one so we aren't initializing then hiding all the others unnecessarily, when separate instances of the page will be open dozens of times on the Rustdoc page inside many iframes.

Reply:
Ah I see. I'll make that change!

Also oops, I didn't see your previous comment when writing mine - I hadn't refreshed the page yet.